rest/model/project.go

```diff
@@ -182,7 +182,7 @@ type APIProjectRef struct {
 	RepotrackerDisabled         *bool                `json:"repotracker_disabled"`
 	DispatchingDisabled         *bool                `json:"dispatching_disabled"`
 	DisabledStatsCache          *bool                `json:"disabled_stats_cache"`
-	FilesIgnoredFromCache       []*string            `json:"files_ignored_from_cache,omitempty"`
+	FilesIgnoredFromCache       []*string            `json:"files_ignored_from_cache"`
```

Comment:
@john-m-liu: part of why I made these omitempty changes is because I think an empty array and a nil array should have different behavior: a nil array means "default to repo", whereas an empty array means "use these values, there just aren't any". It looks like for this change to work, [FromStrPtrSlice](https://github.com/evergreen-ci/evergreen/blob/6fb8d35d10621d702b2dbb66c74eb1ef45bd0671/vendor/github.com/evergreen-ci/utility/optional.go#L168) will need to be written to this: 
```
func FromStringPtrSlice(in []*string) []string {
	if in == nil {
		return nil
	}
	res := []string{}
	for _, each := range in {
		res = append(res, FromStringPtr(each))
	}
	return res
}
```
if this seems reasonable, I'll open a utility PR for it

Reply:
sgtm

Reply:
....I could've sworn that FromStringPtrSlice used to say `res := []string{}` and that's why the test was failing / why I said all of this, but I went back and looked and it totally does what I want, so.... ignore this thread i guess I just won't sleep at night trying to figure out what happened here